### ISSUE TITLE
Request for improvement of typechecker errors

### ISSUE THREAD

**Issue by RieksJ** [AmpersandTarski/Ampersand#1147]
Consider the following script:
~~~
RELATION r[A*B]
--RELATION s[B*C]
RELATION t[X*Y]
RELATION u[Y*A]
RULE "test": I[A] |- r;s;V \/ V;t;u
~~~
This produces the following type-checker error:
~~~
  Cannot disambiguate: V
    Please add a signature (e.g. [A*B]) to it.
  Note: Some cases are not disambiguated fully by design. You can read about
    this at https://github.com/AmpersandTarski/Ampersand/issues/980#issuecomment-508985676 
~~~
Note that this error is caused by relation `s` not being defined (it was commented out to show this).

I've often noted that the cause of a disambiguation error is not that the types of relations are wrong, but that a relation isn't defined. It would be helpful if the typechecker would first check whether or not the relations in the expression are actually defined (throwing an error if that's not the case) before making an attempt to check their types. @sjcjoosten : is that feasible?

**Comment by sjcjoosten** [comment]
Absolutely! I'm a little surprised that only one type-checker error is shown here, but yes fixing this is absolutely desirable. It's probably a bit more than a day's work but consider me on it!

**Comment by hanjoosten** [comment]
I suppose it would be very helpful to list all defined relations that have a name which is used in the expression that is being disambiguated. That shouldn't be too hard to implement.
